lib/moonshot/artifact_repository/s3_bucket_via_github_releases.rb

```diff
@@ -70,13 +77,143 @@ def transfer_release_asset_to_s3(version)
     def github_to_s3(version, s3_name)
       Dir.mktmpdir('github_to_s3', Dir.getwd) do |tmpdir|
         Dir.chdir(tmpdir) do
-          sh_out("hub release download #{version}")
-          file = Dir.glob("*#{version}*.tar.gz").fetch(0)
+          file = download_from_github(version)
           upload_to_s3(file, s3_name)
         end
       end
     end
 
+    # Uploads the file to s3 and verifies the checksum.
+    #
+    # @param file [String] File to be uploaded to s3.
+    # @param key [String] Name of the object to be created on s3.
+    # @raise [RuntimeError] If the file fails to upload correctly after 3
```

Comment:
The scope of RuntimeError usage is beyond this PR. We will fight another day RuntimeError!

 - Don Quixote